src/libopenrave/robot.cpp

```diff
@@ -2747,6 +2537,10 @@ UpdateFromInfoResult RobotBase::UpdateFromRobotInfo(const RobotBaseInfo& info)
 
     // manipulators
     FOREACHC(itManipulatorInfo, info._vManipulatorInfos) {
+        if ((*itManipulatorInfo)->_id.empty()) {
```

Comment:
@ziyan skip updating because no `id`? is that safe? perhaps we should match names..?

Reply:
No longer skipping id.